One sample Z test

Suppose a wildlife biologist is studying the average weight of a particular species of wild rabbits in a forest. The biologist knows from previous studies that the population mean weight of these rabbits is 2.5 kg with a standard deviation of 0.5 kg. The biologist collects a sample of 30 rabbits from a different area of the forest to determine if their average weight significantly differs from the known population mean.n.n.n.n.n.n.n.

In [1]:
import numpy as np
from scipy import stats

In [2]:
# Given data
#Data Input: The weights of the 30 rabbits are stored in the weights list.
weights = [
    2.3, 2.7, 2.6, 2.4, 2.9, 2.8, 2.2, 2.5, 2.3, 2.6,
    2.7, 2.4, 2.5, 2.6, 2.3, 2.4, 2.5, 2.7, 2.6, 2.3,
    2.8, 2.7, 2.5, 2.4, 2.6, 2.2, 2.3, 2.7, 2.5, 2.6
]

In [3]:
# Population parameters
# Population Parameters: The known population mean (2.5 kg) and standard deviation (0.5 kg) are defined.
population_mean = 2.5
population_std_dev = 0.5
sample_size = len(weights)

In [4]:
# Calculate sample mean
# Sample Mean Calculation: The mean of the sample data is calculated using np.mean(weights).
sample_mean = np.mean(weights)


In [5]:
# Calculate the z-score
z_score = (sample_mean - population_mean) / (population_std_dev / np.sqrt(sample_size))

In [6]:
# Calculate the p-value
p_value = stats.norm.sf(abs(z_score)) * 2  # Two-tailed test


In [7]:
# Output results
print(f'Sample Mean: {sample_mean:.2f}')
print(f'Z-Score: {z_score:.2f}')
print(f'P-Value: {p_value:.4f}')

Sample Mean: 2.52
Z-Score: 0.22
P-Value: 0.8266


In [8]:
# Check if the result is significant at alpha = 0.05
alpha = 0.05
if p_value < alpha:
    print("Reject the null hypothesis: There is a significant difference in the average weight.")
else:
    print("Fail to reject the null hypothesis: There is no significant difference in the average weight.")

Fail to reject the null hypothesis: There is no significant difference in the average weight.


The calculations provide the following results:
Sample Mean: 2.52 kg
Z-Score: 0.22
P-Value: 0.8266
Since the p-value (0.8266) is much greater than the common significance level of 0.05, we fail to reject the null hypothesis. This indicates that there is no significant difference in the average weight of the rabbits in this sample compared to the known population mean
